ompi/mca/osc/ucx/osc_ucx_comm.c

```diff
@@ -966,9 +966,7 @@ int ompi_osc_ucx_rget(void *origin_addr, int origin_count,
     rkey = (module->win_info_array[target]).rkey;
 
     OMPI_OSC_UCX_REQUEST_ALLOC(win, ucx_req);
-    if (NULL == ucx_req) {
-        return OMPI_ERR_TEMP_OUT_OF_RESOURCE;
-    }
+    assert(NULL != ucx_req);
```

Comment:
can we return a failure instead of assert? or would need to add proper resource cleanup?